vivisect/__init__.py

```diff
@@ -1111,6 +1112,7 @@ def isProbablyCode(self, va, **kwargs):
     #
     # Opcode API
     #
+    @functools.cache
```

Comment:
i still think there is a danger here with self modifying code.

if the underlying bytes change out from under the cache, then `parseOpcode` will return the wrong insn, and all sorts of things will break.

Reply:
caching using the tuple key `(off, b)` from below seems like it might work and would still accomplish your goal.

Reply:
is the cache object part of the API? should it be exposed somewhere? is it ever reasonable for a user to disable the cache? how about clear it?

Reply:
And that's valid. I wanted to throw this in since I couldn't find our old conversation (and forgot what you had said), but you had some valid concern that I couldn't remember off the top of my head. 

But parseOpcode is hit more than enough that I want to force the conversation for this. And I'm less concerned with how we cache and more conerned that only part of our code does any caching.

Reply:
As for the emulator's caching, that's all ephemeral to each emulator instance, so I'm actually going to remove that and just so with the top level caching instance so that we actually see some form of gain.

As for the API surrounding it, right now it's now. But In caching via bytes or some other more robust mechanism, it would end up being a property on the workspace, so people could reach in and tinker with it should they want to, and I will add a method to just wipe the cache entirely.

Reply:
i'm not sure i understand what you're saying in this last comment.  what are you going to rip out?

Reply:
I just made the emulator be a pass through to the base workspace parseOpcode so we actually use that cache and get the benefit of it.